cli/daemon/dash/dashapp/src/components/Combobox.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { FC, FocusEvent, useEffect, useRef, useState } from "react";
+import fuzzysort from "fuzzysort";
+import { Combobox as HeadlessCombobox } from "@headlessui/react";
+import * as icons from "~c/icons";
+
+const classNames = (...classes: [string, string | boolean]) =>
+  classes.filter(Boolean).join(" ");
+
+export interface ComboboxOptionsItem {
+  name: string;
+}
+
+interface Props {
+  label: string;
+  selectedItem: ComboboxOptionsItem;
+  items: ComboboxOptionsItem[];
+  onChange: (selected: ComboboxOptionsItem) => void;
+}
+
+  const Combobox: FC<Props> = ({ label, selectedItem, onChange, items }) => {
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const [query, setQuery] = useState<string>("");
+
+  useEffect(() => {
+    const handleClickOutside = (event: any) => {
+      if (wrapperRef.current && !wrapperRef.current.contains(event.target)) {
+        setQuery("");
+      }
+    };
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, [wrapperRef]);
+
+  const filteredItems = fuzzysort.go(query, items, {
+    key: "name",
+    all: true,
+    threshold: -200,
+  });
+
+  return (
+    <HeadlessCombobox
```

Comment:
Does the HeadlessCombobox handle aria labels etc? We use them on the webapp, would be nice to start using them everywhere to improve the accessibility

Reply:
Thanks! 
Yes, they add some aria stuff:
![image](https://user-images.githubusercontent.com/1826823/182402638-6d33064b-045e-4b37-89e0-d9fe5b7e4308.png)

They also add the functionality to select via keyboard and tab so it's not mouse/trackpad dependent. 
https://headlessui.com/react/combobox#accessibility-notes

Reply:
Awesome! 🙏